Add "no scope" mod for osu

jaswon opened the pull request:
Resolves https://github.com/ppy/osu/discussions/13082

~"Ghost"~ "No Scope" mod fades out the cursor, resetting on combo break

jaswon committed:
Add "ghost" mod for osu

bdach reviewed (commented):
just from a source read-through, have not tested in any capacity

jaswon committed:
address comments

jaswon committed:
revert interpolation changes + fix cursor trail

jaswon committed:
Rename Mod

jaswon commented:
> 
> 
> Not sure we want to call this "ghost" because the terminology is usually used for a "ghost mode", where you are competing against yourself. This is something we will likely implement in a future, so I think reserving the term makes sense.

This makes sense to me. I've renamed the mod to "No Scope", but welcome any other suggestions

peppy committed:
Use `Interpolation.Lerp` instead of transforms

Better handles cases where the combo may be changing faster than the
transition length.

peppy committed:
Make const private

peppy commented:
Looks good. Not sure about the icon but it will change later down the path.
